src/Backend/Modules/MediaGalleries/Domain/MediaGallery/MediaGallery.php

```diff
@@ -0,0 +1,403 @@
+     */
+    public function getUserId(): int
+    {
+        return $this->userId;
+    }
+
+    /**
+     * Gets the value of moduleExtraId.
+     *
+     * @return integer
+     */
+    public function getModuleExtraId(): int
+    {
+        return $this->moduleExtraId;
+    }
+
+    /**
+     * Gets the Action
+     *
+     * @return string
+     */
+    public function getAction(): string
+    {
+        return $this->action;
+    }
+
+    /**
+     * Gets the value of title.
+     *
+     * @return string
+     */
+    public function getTitle(): string
+    {
+        return $this->title;
+    }
+
+    /**
+     * Gets the value of text.
+     *
+     * @return string|null
+     */
+    public function getText()
+    {
+        return $this->text;
+    }
+
+    /**
+     * Gets the value of createdOn.
+     *
+     * @return \DateTime
+     */
+    public function getCreatedOn(): \DateTime
+    {
+        return $this->createdOn;
+    }
+
+    /**
+     * Gets the value of editedOn.
+     *
+     * @return \DateTime
+     */
+    public function getEditedOn(): \DateTime
+    {
+        return $this->editedOn;
+    }
+
+    /**
+     * Gets the value of publishOn.
+     *
+     * @return \DateTime
+     */
+    public function getPublishOn(): \DateTime
+    {
+        return $this->publishOn;
+    }
+
+    /**
+     * @return Status
+     */
+    public function getStatus(): Status
+    {
+        return $this->status;
+    }
+
+    /**
+     * Gets the value of mediaGroup.
+     *
+     * @return MediaGroup
+     */
+    public function getMediaGroup(): MediaGroup
+    {
+        return $this->mediaGroup;
+    }
+
+    /**
+     * @ORM\PrePersist
+     */
+    public function onPrePersist()
+    {
+        $this->createdOn = $this->editedOn = new \Datetime();
+
+        $this->moduleExtraId = Model::insertExtra(
+            ModuleExtraType::widget(),
+            'MediaGalleries',
+            'Gallery',
+            'Gallery',
+            array(),
+            false
+        );
+    }
+
+    /**
+     * @ORM\PreUpdate
+     */
+    public function onPreUpdate()
+    {
+        $this->editedOn = new \Datetime();
+    }
+
+    /**
+     * @ORM\PostPersist
+     */
+    public function onPostPersist()
+    {
+        $this->updateModuleExtraData();
+    }
+
+    /**
+     * @ORM\PostUpdate
+     */
+    public function onPostUpdate()
+    {
+        $this->updateModuleExtraData();
+    }
+
+    /**
+     * @ORM\PostRemove
+     */
+    public function onPostRemove()
+    {
+        Model::deleteExtraById(
+            $this->moduleExtraId,
+            true
+        );
+    }
+
+    /**
+     * Get extra label
+     *
+     * @return string The gallery extra_label.
+     */
+    protected function getExtraLabel(): string
+    {
+        return '"' . $this->title . '"'
+            . ' - '
+            . ucfirst($this->action);
+    }
+
+    /**
+     * Update module extra data
+     */
+    protected function updateModuleExtraData()
```

Comment:
why don't you just make this one public and add the annotations for post persist and update on this method, that is way clearer than a method postPersist that calls this one